core/cmd/local_client.go

```diff
@@ -527,7 +528,42 @@ func (cli *Client) PrepareTestDatabase(c *clipkg.Context) error {
 		return cli.errorOut(err)
 	}
 
-	return nil
+	return cli.errorOut(dropDanglingTestDBs(cli.Logger, db))
+}
+
+func dropDanglingTestDBs(lggr logger.Logger, db *sqlx.DB) (err error) {
+	// Drop all old dangling databases
+	var dbs []string
+	if err = db.Select(&dbs, `SELECT datname FROM pg_database WHERE datistemplate = false;`); err != nil {
+		return err
+	}
+
+	// dropping database is very slow in postgres so we parallelise it here
+	nWorkers := 25
+	ch := make(chan string)
+	wg := sync.WaitGroup{}
+	wg.Add(nWorkers)
+	errMu := sync.Mutex{}
```

Comment:
nit/ I thought the linter complained about these? Maybe only for slice: `s := []int{}` :thinking: 
```suggestion
	var wg sync.WaitGroup
	wg.Add(nWorkers)
	var errMu sync.Mutex
```